src/main/java/org/veupathdb/service/eda/merge/core/MergeRequestProcessor.java

```diff
@@ -49,71 +55,66 @@ public MergeRequestProcessor(MergedTabularRequestResources resources) {
+    Study study = Resources.getStudyResolver().getStudyById(_resources.getMetadata().getStudyId());
 
     // request validated; convert requested entity and vars to defs
     EntityDef targetEntity = metadata.getEntity(targetEntityId).orElseThrow();
     List<VariableDef> outputVarDefs = metadata.getTabularColumns(targetEntity, outputVarSpecs);
-    List<VariableSpec> outputVars = new ArrayList<>(outputVarDefs.stream().map(v -> (VariableSpec)v).toList());
 
     // build entity node tree to aggregate the data into a streaming response
     RootStreamingEntityNode targetStream = new RootStreamingEntityNode(targetEntity, outputVarDefs,
         _resources.getSubsetFilters(), metadata, _resources.getDerivedVariableFactory(), computeInfo);
     LOG.info("Created the following entity node tree: " + targetStream);
 
+    // Use metadata cache directly, which bypasses user studies, since user studies don't currently have files.
+    final boolean fileBasedSubsetting = Resources.getMetadataCache().studyHasFiles(study.getStudyId());
+
     // get stream specs for streams needed by the node tree, which will be merged into this request's response
     Map<String, StreamSpec> requiredStreams = Functions.getMapFromValues(targetStream.getRequiredStreamSpecs(), StreamSpec::getStreamName);
 
     // create stream generator
-    Function<StreamSpec, ResponseFuture> streamGenerator = spec ->
+    Function<StreamSpec, CloseableIterator<Map<String, String>>> streamGenerator = spec ->
         COMPUTED_VAR_STREAM_NAME.equals(spec.getStreamName())
-        // need to get compute stream from compute service
-        ? _resources.getComputeTabularStream()
-        // all other streams come from subsetting service
-        : _resources.getSubsettingTabularStream(spec);
+            // need to get compute stream from compute service
+            ? _resources.getComputeStreamIterator(study)
+            // all other streams come from subsetting service
+            : FilteredResultFactory.tabularSubsetIterator(study,
+            study.getEntity(spec.getEntityId()).orElseThrow(),
+            getVariablesFromStreamSpec(spec, study),
+            ApiConversionUtil.toInternalFilters(study, spec.getFiltersOverride().orElse(_resources.getSubsetFilters()), Resources.getAppDbSchema()),
+            Resources.getBinaryValuesStreamer(), fileBasedSubsetting, Resources.getApplicationDataSource(), Resources.getAppDbSchema());
 
     return out -> {
 
       // create stream processor
-      ConsumerWithException<Map<String,InputStream>> streamProcessor =
-          targetStream.requiresNoDataManipulation()
-          ? dataStreams -> writePassThroughStream(outputVars, dataStreams.values().iterator().next(), out)
-          : dataStreams -> writeMergedStream(targetStream, dataStreams, out);
+      ConsumerWithException<Map<String, CloseableIterator<Map<String, String>>>> streamProcessor =
+          dataStreams -> writeMergedStream(targetStream, dataStreams, out);
```

Comment:
I could add it back, the main reason it was there in the first place was to avoid deserializing from bytes -> `Map<String, String>` but now our stream is already Java objects it doesn't provide much a performance benefit and it adds some complexity.